### PR TITLE
scylla_ntp_setup: support systemd-timesyncd

### LINKED ISSUE #8339: scylla_ntp_setup: Use systemd-timesyncd instead of chrony when it's installed by default
systemd added NTP client since systemd 213, called systemd-timesyncd.
Seems like RHEL8/CentOS8 disabling it, but on Ubuntu it's the default installed NTP client.
To skip installing unnecessary package, and follow distribution default configuration, use systemd-timesyncd instead of chrony on Ubuntu.

See: https://ubuntu.com/server/docs/network-ntp

It says:

_Since Ubuntu 16.04 timedatectl / timesyncd (which are part of systemd) replace most of ntpdate / ntp.
timesyncd is available by default and replaces not only ntpdate, but also the client portion of chrony (or formerly ntpd)._

### PR DESCRIPTION
**Pull request by syuu1228**
On Ubuntu/Debian systemd-timesyncd is default NTP client, and installed
by default.
So use it instead of installing chrony.

Fixes #8339

### CODE REVIEW

**Comment by yaronkaikov** [comment]
@syuu1228 I am facing the same issue while working on creating Scylla Azure image based on ubnutu20.04 , tried to use your fix, but it failed.
```
==> azure-arm: See "systemctl status chrony.service" and "journalctl -xe" for details.
==> azure-arm: Traceback (most recent call last):
==> azure-arm:   File "/opt/scylladb/scripts/libexec/scylla_ntp_setup", line 62, in <module>
    azure-arm: NTP setup failed.
```

Shouldn't it use systemd-timesyncd by default?

File: dist/common/scripts/scylla_ntp_setup [hunk 1]
```diff
@@ -40,24 +40,29 @@ if __name__ == '__main__':
...
 
-    if shutil.which('ntpd'):
-        pkg_uninstall('ntp')
-    if not shutil.which('chronyd'):
-        pkg_install('chrony')
-    if is_debian_variant():
-        confpath = '/etc/chrony/chrony.conf'
-        chronyd = 'chrony.service'
+    # use systemd-timesyncd
+    if systemd_unit.available('systemd-timesyncd'):
```
Comment (yaronkaikov) [inline dist/common/scripts/scylla_ntp_setup:44]: i think you are missing an import of systemd_unit here
and AFAIK it's not installed,so we should install it first

**Comment by syuu1228** [comment]
> @syuu1228 I am facing the same issue while working on creating Scylla Azure image based on ubnutu20.04 , tried to use your fix, but it failed.
> 
> ```
> ==> azure-arm: See "systemctl status chrony.service" and "journalctl -xe" for details.
> ==> azure-arm: Traceback (most recent call last):
> ==> azure-arm:   File "/opt/scylladb/scripts/libexec/scylla_ntp_setup", line 62, in <module>
>     azure-arm: NTP setup failed.
> ```
> 
> Shouldn't it use systemd-timesyncd by default?

I thought systemd-timesyncd is installed by default on these distributions (at least it installed on Vagrant Ubuntu image).
If Azure's (or other IaaS providers) Ubuntu image does not installed it by default, we need to change this patch logic.
Currently, it works like this:
```
if systemd-timesyncd is installed:
    setup systemd-timesyncd
else
    install chrony
    setup chrony
```
Instead of this logic, we need to install systemd-timesyncd when distribution is not CentOS/RHEL:
```
if distro.id_like != rhel:
    install systemd-timesyncd
    setup systemd-timesyncd
else
    install chrony
    setup chrony
```

File: dist/common/scripts/scylla_ntp_setup (hunk 1 above)
Comment (syuu1228) [reply dist/common/scripts/scylla_ntp_setup:44]: systemd_unit is imported at `from scylla_util import *`, and it always usable across scylla setup scripts.

**Comment by syuu1228** [comment]
Or, maybe 
```
try:
    pkg_install('systemd-timesyncd')
    mode = 'systemd-timesyncd'
except:
    pass
if not mode:
    pkg_install('chrony')
    mode = 'chrony'
```
I guess it's better to specify distribution, because this may works on more distributions (which is unsupported).

**Comment by syuu1228** [comment]
I guess we need a code to skip ntp setup when it's already configured, anyway.
Or, we can add --no-ntp-setup in scylla_image_setup to do so.

**Comment by avikivity** [comment]
Best would be to detect what is installed and use it. All modern systems should have something.

Maybe we can replace installation with a warning that you don't have a time server setup and that you should get one.

**Comment by avikivity** [comment]
(assuming I'm correct and that modern systems not only install but also enable a time server)

**Comment by yaronkaikov** [comment]
@syuu1228 your patch is working . i tested it before the wrong way.

**Review by yaronkaikov — APPROVED** [review]

**Comment by slivne** [comment]
@avikivity / @penberg this is blocking Azure Images

**Comment by penberg** [comment]
@avikivity I merged this as is, because it unblocks Azure image. @syuu1228, I think @avikivity's suggestion of replacing installation with a warning makes sense, though.

**Pull request closed without merging**